http_logs/challenges/common/default-schedule.json

```diff
@@ -68,30 +68,64 @@
           "iterations": 100,
           "target-throughput": 8
         },
+{%- if runtime_fields is defined %}
+        { "operation": "term", "name": "term-from-source-using-grok",       "warmup-iterations": 50, "iterations": 10, "target-throughput": 0.01 },
```

Comment:
One idea here is to label similar tasks with the [tags](https://esrally.readthedocs.io/en/stable/track.html?highlight=tag#schedule) property.

Let me explain the usefulness. Say for the nightlies we wanna have a run with only `*-using-index-of`.

Now we'd have to something like:

`esrally race ... --exclude-tasks="term-from-source-using-grok,term-from-keyword-using-grok,term-from-wildcard-using-grok,term-from-source-using-dissect,term-from-keyword-using-dissect,term-from-wildcard-using-dissect,200s-in-range-from-source-using-grok,200s-in-range-from-keyword-using-grok,200s-in-range-from-wildcard-using-grok,200s-in-range-from-source-using-dissect,200s-in-range-from-keyword-using-dissect,200s-in-range-from-wildcard-using-dissect,400s-in-range-from-source-using-grok,400s-in-range-from-keyword-using-grok,400s-in-range-from-wildcard-using-grok,400s-in-range-from-source-using-dissect,400s-in-range-from-keyword-using-dissect,400s-in-range-from-wildcard-using-dissect"`

To simply this, if we'd specify the tasks like:

```
        { "tags": "grok", "operation": "200s-in-range", "name": "200s-in-range-from-source-using-grok",       "warmup-iterations": 500, "iterations": 100, "target-throughput": 5 },
        { "tags": "grok", "operation": "200s-in-range", "name": "200s-in-range-from-keyword-using-grok",      "warmup-iterations": 500, "iterations": 100, "target-throughput": 5 },
        { "tags": "grok", "operation": "200s-in-range", "name": "200s-in-range-from-wildcard-using-grok",     "warmup-iterations": 500, "iterations": 100, "target-throughput": 5 },
        { "tags": "dissect", "operation": "200s-in-range", "name": "200s-in-range-from-source-using-dissect",    "warmup-iterations": 500, "iterations": 100, "target-throughput": 5 },
        { "tags": "dissect", "operation": "200s-in-range", "name": "200s-in-range-from-keyword-using-dissect",   "warmup-iterations": 500, "iterations": 100, "target-throughput": 5 },
        { "tags": "dissect", "operation": "200s-in-range", "name": "200s-in-range-from-wildcard-using-dissect",  "warmup-iterations": 500, "iterations": 100, "target-throughput": 5 },
        { "tags": "index-of", "operation": "200s-in-range", "name": "200s-in-range-from-source-using-index-of",   "warmup-iterations": 500, "iterations": 100, "target-throughput": 5 },
        { "tags": "index-of", "operation": "200s-in-range", "name": "200s-in-range-from-keyword-using-index-of",  "warmup-iterations": 500, "iterations": 100, "target-throughput": 5 },
        { "tags": "index-of", "operation": "200s-in-range", "name": "200s-in-range-from-wildcard-using-index-of", "warmup-iterations": 500, "iterations": 100, "target-throughput": 5 },
        { "tags": "grok", "operation": "400s-in-range", "name": "400s-in-range-from-source-using-grok",       "warmup-iterations": 500, "iterations": 100, "target-throughput": 0.3 },
        { "tags": "grok", "operation": "400s-in-range", "name": "400s-in-range-from-keyword-using-grok",      "warmup-iterations": 500, "iterations": 100, "target-throughput": 0.3 },
        { "tags": "grok", "operation": "400s-in-range", "name": "400s-in-range-from-wildcard-using-grok",     "warmup-iterations": 500, "iterations": 100, "target-throughput": 0.3 },
        { "tags": "dissect", "operation": "400s-in-range", "name": "400s-in-range-from-source-using-dissect",    "warmup-iterations": 500, "iterations": 100, "target-throughput": 0.3 },
        { "tags": "dissect", "operation": "400s-in-range", "name": "400s-in-range-from-keyword-using-dissect",   "warmup-iterations": 500, "iterations": 100, "target-throughput": 0.3 },
        { "tags": "dissect", "operation": "400s-in-range", "name": "400s-in-range-from-wildcard-using-dissect",  "warmup-iterations": 500, "iterations": 100, "target-throughput": 0.3 },
        { "tags": "index-of", "operation": "400s-in-range", "name": "400s-in-range-from-source-using-index-of",   "warmup-iterations": 500, "iterations": 100, "target-throughput": 0.3 },
        { "tags": "index-of", "operation": "400s-in-range", "name": "400s-in-range-from-keyword-using-index-of",  "warmup-iterations": 500, "iterations": 100, "target-throughput": 0.3 },
        { "tags": "index-of", "operation": "400s-in-range", "name": "400s-in-range-from-wildcard-using-index-of", "warmup-iterations": 500, "iterations": 100, "target-throughput": 0.3 },
```

and similarly below for the `200s-in-range` and `400s-in-range` we could then filter using `--exclude-tasks="tag:grok,tag:dissect"; see this example with the above modification:

```
17:25 $ esrally info --track-path=/path/to/http_logs --challenge=runtime-fields --track-params='runtime_fields:true' --exclude-tasks="tag:grok,tag:dissect"

    ____        ____
   / __ \____ _/ / /_  __
  / /_/ / __ `/ / / / / /
 / _, _/ /_/ / / / /_/ /
/_/ |_|\__,_/_/_/\__, /
                /____/

Showing details for track [http_logs]:

* Description: HTTP server log data
* Documents: 247,249,096
* Compressed Size: 1.1 GB
* Uncompressed Size: 26.0 GB

==========================
Challenge [runtime-fields]
==========================

Indexes the whole document corpus using scripts to extract fields. Set the track param `runtime_fields` to `true`.

Schedule:
----------

1. delete-index
2. create-index
3. check-cluster-health
4. create-timestamp-pipeline
5. index-append-with-timestamp-pipeline (8 clients)
6. refresh-after-index
7. force-merge
8. refresh-after-force-merge
9. wait-until-merges-finish
10. default
11. term-from-source-using-index-of
12. term-from-keyword-using-index-of
13. term-from-wildcard-using-index-of
14. range
15. 200s-in-range-from-source-using-index-of
16. 200s-in-range-from-keyword-using-index-of
17. 200s-in-range-from-wildcard-using-index-of
18. 400s-in-range-from-source-using-index-of
19. 400s-in-range-from-keyword-using-index-of
20. 400s-in-range-from-wildcard-using-index-of
21. hourly_agg
22. scroll
23. desc_sort_timestamp
24. asc_sort_timestamp
25. desc_sort_with_after_timestamp
26. asc_sort_with_after_timestamp
27. force-merge-1-seg
28. refresh-after-force-merge-1-seg
29. wait-until-merges-1-seg-finish
30. desc-sort-timestamp-after-force-merge-1-seg
31. asc-sort-timestamp-after-force-merge-1-seg
32. desc-sort-with-after-timestamp-after-force-merge-1-seg
33. asc-sort-with-after-timestamp-after-force-merge-1-seg

-------------------------------
[INFO] SUCCESS (took 0 seconds)
-------------------------------
```

this could improve usability a lot.

Reply:
Nice! I didn't know about those.